app/src/main/java/com/android/sample/model/preferences/PreferencesLocalCache.kt

```diff
@@ -0,0 +1,45 @@
+package com.android.sample.model.preferences
+
+import android.content.Context
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.emptyPreferences
+import androidx.datastore.preferences.core.stringPreferencesKey
+import androidx.datastore.preferences.preferencesDataStore
+import com.google.gson.Gson
+import java.io.IOException
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.catch
+import kotlinx.coroutines.flow.map
+
+val Context.preferencesDataStore: DataStore<Preferences> by
+    preferencesDataStore(name = "preferences_cache")
+
+class PreferencesLocalCache(private val context: Context) {
+  private val gson = Gson()
+  private val preferencesKey = stringPreferencesKey("preferences_data")
```

Comment:
Maybe you could add Kdoc here.

Reply:
I removed this and will include it in a second PR specific to the Preferences but the Kdoc has been added there